packages/toolpad-app/src/components/DialogForm.tsx

```diff
@@ -1,7 +1,36 @@
+import React from 'react';
 import { styled } from '@mui/material';
 
-export default styled('form')({
+const StyledDialogForm = styled('form')({
   overflowY: 'auto',
   display: 'flex',
   flexDirection: 'column',
 });
+
+export interface DialogFormProps extends React.FormHTMLAttributes<HTMLFormElement> {
+  children: React.ReactNode;
+  onKeyDown?: (event: React.KeyboardEvent) => void;
+}
+
+export default ({ children, onKeyDown = () => {}, ...rest }: DialogFormProps) => {
+  const handleKeyDown = React.useCallback((event: React.KeyboardEvent) => {
```

Comment:
I think that it would be better without the useCallback. There isn't performance concerns here form what I understand.

Reply:
I'm using that as per Jan's suggestion in some previous PRs

Reply:
> There is no performance concerns here from what I understand.

You often need implementation details to judge whether it'll be a performance concern or not. Always passing stable values errs on the side of caution. In this case the implementation of the component is very local to the consumer and we can see there are no performance concerns. I'm fine with doing it either way in this specific case. Especially if there's a chance the component will be extracted into its own file at some point it could be interesting to be defensive about this.

Reply:
I don't think that implementation details matter, IMO the only thing that matters in regards to performance optimisation is measuring and being sure that it is a problem and applying necessary optimisations.

> In this case the implementation of the component is very local to the consumer and we can see there are no performance concerns.

I don't think we can SEE, unless we measure. I'd argue that this is true for majority of the cases where we use this (or useMemo) optimisation. IMO unless we measure the performance it is really subjective to claim whether it's needed or not. I'd personally prefer if it was either used everywhere blindly or only in the places that are truly problematic and needs optimising, otherwise it can just lead to unnecessary speculation and discussions whether THIS IS the place where we need to use optimisation or not.

I'd add that in general I wouldn't be against being "defensive" about this, but react optimisations do make code look more cluttered/ugly than it could be, so I'd prefer to avoid that.